src/hmem_cuda.c

```diff
@@ -625,17 +626,23 @@ int cuda_hmem_init(void)
 		}
 	}
 
-	ret = 1;
-	fi_param_get_bool(NULL, "hmem_cuda_enable_xfer", &ret);
-	cuda_enable_xfer = (ret != 0);
+	param_value = 1;
+	ret = fi_param_get_bool(NULL, "hmem_cuda_enable_xfer", &param_value);
+	if (ret == -FI_ENODATA) {
+		hmem_cuda_xfer_setting = CUDA_XFER_UNSPECIFIED;
+	} else {
+		hmem_cuda_xfer_setting = (param_value > 0) ? CUDA_XFER_ENABLED : CUDA_XFER_DISABLED;
+	}
```

Comment:
It would be safer to write this as:
```
if (!ret) {
    hmem_cuda_xfer_setting = ...
}
```
fi_param_get can return other failure codes (EINVAL, ENOENT) -- neither of those should happen.

Reply:
Updated to 

```
         if (ret) {
		hmem_cuda_xfer_setting = CUDA_XFER_UNSPECIFIED;
	} else {
		hmem_cuda_xfer_setting = (param_value > 0) ? CUDA_XFER_ENABLED : CUDA_XFER_DISABLED;
	}
```

as suggested.